core/libp2p/host.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/host.hpp"
+
+namespace libp2p {
+
+  peer::PeerId Node::getId() const noexcept {
+    return id_;
+  }
+
+  gsl::span<const multi::Multiaddress> Node::getListenAddresses() {
+    // TODO
+  }
+
+  peer::PeerRepository &Node::getPeerRepository() const {
+    return *config_.peer_repository;
+  }
+
+  outcome::result<void> Node::addKnownPeer(
+      const peer::PeerId &p, gsl::span<const multi::Multiaddress> mas,
+      std::chrono::milliseconds ttl) {
+    return config_.peer_repository->getAddressRepository().upsertAddresses(
+        p, mas, ttl);
+  }
+
+  outcome::result<void> Node::newStream(
+      const peer::PeerId &p, const peer::Protocol &protocol,
+      const std::function<void(std::unique_ptr<stream::Stream>)> &handler) {
+    // TODO(@warchant):
```

Comment:
I hope you are going to clarify what it all means later )

Reply:
These are some notes for the one, who will implement this algorithm. When they are converted to code, yes, I hope, they will become clearer